drivers/sensor/ltrf216a/ltrf216a.c

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2023 Tridonic
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ltr_f216a
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/__assert.h>
+
+#include "ltrf216a.h"
+
+LOG_MODULE_REGISTER(ltrf216a, CONFIG_SENSOR_LOG_LEVEL);
+
+/*
+ * Driver for I2C illuminance sensor LiteOn LTR-F216A.
+ *
+ * Datasheet:
+ * <https://optoelectronics.liteon.com/upload/download/DS86-2019-0016/LTR-F216A_Final_DS_V1.4.PDF>
+
+ * 7bit Address 0x53
+ * 8bit Address 0xA6 read
+ * 8bit Address 0xA7 write
+
+ * NOT IMPLEMENTED:
+ * - Interrupt
+ * - Modifying Gain (using default x3)
+ * - Modifying Resolution (using default 100ms)
+ * - Modifying Measurement Rate (using default 100ms)
+ * - Modifying Window Factor (using default 1)
+ */
+
+#define LTRF216A_ALS_RESET_MASK  16
+#define LTRF216A_ALS_ENABLE_MASK 2
+
+#define LTRF216A_ALS_DATA_STATUS 8
+
+/* Part Number ID 7:4 0b1011 */
+#define LTRF216A_PART_ID_PARTNUMBERID_VALUE 0xB
+#define LTRF216A_PART_ID_PARTNUMBERID_MASK  0xF0
+#define LTRF216A_PART_ID_PARTNUMBERID_SHIFT 4
+
+/* Revision ID 3:0 0b0001 */
+#define LTRF216A_PART_ID_REVISIONID_VALUE 0x1
+#define LTRF216A_PART_ID_REVISIONID_MASK  0x0F
+
+#define LTRF216A_MAIN_CTRL        0x00
+#define LTRF216A_ALS_MEAS_RES     0x04
+#define LTRF216A_ALS_GAIN         0x05
+#define LTRF216A_PART_ID          0x06
+#define LTRF216A_MAIN_STATUS      0x07
+#define LTRF216A_ALS_CLEAR_DATA_0 0x0a
+#define LTRF216A_ALS_CLEAR_DATA_1 0x0b
+#define LTRF216A_ALS_CLEAR_DATA_2 0x0c
+#define LTRF216A_ALS_DATA_0       0x0d
+#define LTRF216A_ALS_DATA_1       0x0e
+#define LTRF216A_ALS_DATA_2       0x0f
+#define LTRF216A_INT_CFG          0x19
+#define LTRF216A_INT_PST          0x1a
+#define LTRF216A_ALS_THRES_UP_0   0x21
+#define LTRF216A_ALS_THRES_UP_1   0x22
+#define LTRF216A_ALS_THRES_UP_2   0x23
+#define LTRF216A_ALS_THRES_LOW_0  0x24
+#define LTRF216A_ALS_THRES_LOW_1  0x25
+#define LTRF216A_ALS_THRES_LOW_2  0x26
+
+#define LTRF216A_ALS_READ_DATA_DELAY_MS 20
+
+#define LTRF216A_WIN_FAC 1
+
+#define LTRF216A_NUMBER_DATA_REGISTERS 6
+
+static int ltrf216a_reg_read(const struct device *dev, uint8_t reg, uint8_t *val)
+{
+	const struct ltrf216a_config *config = dev->config;
+
+	if (i2c_burst_read_dt(&config->i2c, reg, val, 1) != 0) {
```

Comment:
i2c_burst_read_dt already returns -EIO on failure.
You could return it directly without the if/else, or just use that function directly without this helper.

Reply:
OK, will change this.

Reply:
Do you really even need to wrap I2C functions at all? Just call them directly

Reply:
changed in the latest force push.